src/AppBundle/Entity/Post.php

```diff
@@ -133,13 +133,13 @@ public function setAuthorEmail($authorEmail)
     /**
      * Is the given User the author of this Post?
      *
-     * @param User $user
+     * @param User|string $user
```

Comment:
This is the original code:

```php
    public function isAuthor(User $user = null)
    {
        return $user->getEmail() == $this->getAuthorEmail();
    }
```

Which generates this error:

![before](https://cloud.githubusercontent.com/assets/73419/8804957/f32e9ac2-2fce-11e5-995d-b80e3f82f056.png)

And this is the new code:

```php
    public function isAuthor($user = null)
    {
        return $user instanceof User && $user->getEmail() == $this->getAuthorEmail();
    }
```

Which correctly redirects to the login form:

![after](https://cloud.githubusercontent.com/assets/73419/8804968/ff8187c6-2fce-11e5-9bb3-e72c20f4c1c0.png)